hbase-server/src/main/java/org/apache/hadoop/hbase/snapshot/SnapshotManifest.java

```diff
@@ -317,26 +318,19 @@ protected void addRegion(final Path tableDir, final RegionInfo regionInfo, Regio
       // in batches and may miss files being added/deleted. This could be more robust (iteratively
       // checking to see if we have all the files until we are sure), but the limit is currently
       // 1000 files/batch, far more than the number of store files under a single column family.
-      Collection<String> familyNames = regionFs.getFamilies();
-      if (familyNames != null) {
-        for (String familyName: familyNames) {
-          Object familyData = visitor.familyOpen(regionData, Bytes.toBytes(familyName));
-          monitor.rethrowException();
-
-          Collection<StoreFileInfo> storeFiles = regionFs.getStoreFiles(familyName);
-          if (storeFiles == null) {
-            if (LOG.isDebugEnabled()) {
-              LOG.debug("No files under family: " + familyName);
-            }
-            continue;
-          }
-
-          // 2.1. build the snapshot reference for the store
-          // iterate through all the store's files and create "references".
-          addReferenceFiles(visitor, regionData, familyData, storeFiles, false);
-
-          visitor.familyClose(regionData, familyData);
+      for (ColumnFamilyDescriptor cfd : htd.getColumnFamilies()) {
+        Object familyData = visitor.familyOpen(regionData, cfd.getName());
+        monitor.rethrowException();
+        StoreFileTracker tracker = StoreFileTrackerFactory.create(conf, htd, cfd, regionFs);
```

Comment:
Similar to my above comment. We are assuming conf has the proper store file tracker impl config, but this might be defined at table or CF level.